File: src/main/java/com/abysalto/middleware/exception/InvalidRequestException.java
package com.abysalto.middleware.exception;

/**
 * Thrown for a semantically invalid request that Bean Validation annotations cannot express on their
 * own (e.g. {@code minPrice > maxPrice}). Rendered as a 400 by {@code GlobalExceptionHandler}.
 *
 * <p>Using a dedicated type — rather than a raw {@link IllegalArgumentException} — keeps the 400
 * mapping intentional: an unrelated {@code IllegalArgumentException} thrown deeper in the stack is a
 * server-side fault and correctly surfaces as a 500 instead of being mislabelled a bad request.
 */
public class InvalidRequestException extends RuntimeException {

    public InvalidRequestException(String message) {
        super(message);
    }
}

File: dotnet/src/Middleware.Core/Exceptions/InvalidRequestException.cs
namespace Middleware.Core.Exceptions;

/// <summary>
/// Thrown for a semantically invalid request that field-level validation cannot express on its own
/// (e.g. <c>minPrice &gt; maxPrice</c>). Rendered as a 400. Using a dedicated type keeps the 400
/// mapping intentional: an unrelated argument fault deeper in the stack correctly surfaces as a 500.
/// </summary>
public sealed class InvalidRequestException(string message) : Exception(message);
